sourcegraph/sourcegraph
# [Search jobs] Integrate search jobs UI with api

## Todo 
- [x] Add search job logs UI 
- [x] Connect create search job button to the actual gql mutation 
- [x] Use site admins schema for the searchJobs flag  
- [x] Adjust zero state 
- [x] Add debounce logic to the search and user picker search terms
- [x] Connect delete/restart buttons to the actual gql mutations 
- [x] Enable polling mechanism for the search jobs list and search job logs queries 
- [x] Hide users filter and creator column for non-admin users
- [ ] Add telemetry events (will be added later when RFC and backend changes will be in place)

## BE Todo 
Will be resolved in the following PR https://github.com/sourcegraph/sourcegraph/pull/56640

- [ ] Add usersIds field to the API
- [ ] Add telemetry event SQL upload logic 
- [ ] Add search job logs API 

## Test plan
- Run local instance of sg with `sg start`
- Set seachJobs:true in the site config editor (in the `experimentalFeatures` section)
- Check that you can crate search job from the search result page (though result menu)
- Check that you can observe search jobs list on the search jobs list page `/search-jobs` page (you can navigate to this page through global nav menu, under search dropdown menu item) 
- Check that you can see logs UI when you click search 'View logs' button
- Check that you can delete search job from the search jobs list page
- Check that you can rerun search job by clicking refresh button in the search job row 
- Check that you can download search job result archive on the search job list page
